python/EC2_INSTANCE_NO_PUBLIC_IP/EC2_INSTANCE_NO_PUBLIC_IP.py

```diff
@@ -0,0 +1,365 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+"""
+#####################################
+##           Gherkin               ##
+#####################################
+Rule Name:
+  EC2_INSTANCE_NO_PUBLIC_IP
+Description:
+  Checks whether Amazon EC2 instances have a public IP association or not. The rule is NON_COMPLIANT if the publicIp field is present in the Amazon EC2 instance configuration item. This rule applies only to IPv4.
+Trigger:
+  Configuration Change on AWS::EC2::Instance
+Reports on:
+  AWS::EC2::Instance
+Rule Parameters:
+  None
+Scenarios:
+  Scenario: 1
+     Given: The publicIp field is present in the Amazon EC2 instance configuration item.
+     Then: Return NON_COMPLIANT
+  Scenario: 2
+     Given: The publicIp field is not present in the Amazon EC2 instance configuration item.
+     Then: Return COMPLIANT
+"""
+
+import re
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::EC2::Instance'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    if not re.findall('\\bpublicIp\\b', str(configuration_item)):

```

Comment:
What if a tag is call `publicIp` or something else. This is need to be more elegant.